### PR TITLE
sql: TestValidationWithProtectedTS is skipped 

### LINKED ISSUE #90879: sql: TestValidationWithProtectedTS is flaky
Saw it on like 5 failures on staging branch. E.g. [here](https://teamcity.cockroachdb.com/viewLog.html?buildId=7196927&buildTypeId=Cockroach_UnitTests_BazelUnitTests).
```
No pass/skip/fail event found for test
=== RUN   TestValidationWithProtectedTS
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/7786b2439f6fbd6878e73b0041b1a786/logTestValidationWithProtectedTS808938841
    test_log_scope.go:79: use -show-logs to present logs inline
panic: test timed out after 59m55s
```
Will skip.

Jira issue: CRDB-20977

### PR DESCRIPTION
**Pull request by fqazi**
This patch does the following to address a flaky test:

1. TestValidationWithProtectedTS is re-enabled, and the logic for detecting errors is updated to prevent the test from hanging and handling transaction retry errors.
2. The index backfiller is updated to add a protected timestamp, since it's possible in this test to encounter hangs where we get stuck in retry loops because of short GC interval

Fixes: https://github.com/cockroachdb/cockroach/issues/90879

### CODE REVIEW

File: pkg/sql/schemachanger/scexec/exec_backfill.go [hunk 1]
```diff
@@ -324,7 +326,7 @@ func runBackfiller(
 	im := deps.IndexMerger()
 	const op = "run backfills and merges"
 	bf := func(ctx context.Context, p *BackfillProgress) error {
-		return runBackfill(ctx, deps.IndexSpanSplitter(), ib, *p, tracker, tables[p.TableID])
+		return runBackfill(ctx, deps.IndexSpanSplitter(), ib, *p, tracker, job, tables[p.TableID])
 	}
 	mf := func(ctx context.Context, p *MergeProgress) error {
 		return im.MergeIndexes(ctx, *p, tracker, tables[p.TableID])
```
Comment (postamar) [inline pkg/sql/schemachanger/scexec/exec_backfill.go:332]: Should we also plumb the job in here, to be passed on to a `TryToProtectBeforeGC` call? At this point in this branch we have these calls for all validations and for backfills but not for merges, and merges don't seem special to me.

**Review by postamar — APPROVED** [review]
LGTM, though I understand very little about protected timestamps themselves. Consider getting a second opinion if possible, if this is anything other than low-risk (it doesn't seem particularly risky to me but I don't actually know)

File: pkg/sql/schemachanger/scexec/exec_backfill.go (hunk 1 above)
Comment (fqazi) [reply pkg/sql/schemachanger/scexec/exec_backfill.go:332]: Yes good point we should cover all cases with a AOST queries

**Pull request merged**